groups/ntc/ntcr/ntcr_streamsocket.cpp

```diff
@@ -3679,6 +3790,93 @@ ntsa::Error StreamSocket::privateRetryConnectToEndpoint(
     return ntsa::Error();
 }
 
+ntsa::Error StreamSocket::privateTimestampOutgoingData(bool enable)
+{
+    ntsa::SocketOption option;
+    option.makeTimestampOutgoingData(enable);
+
+    ntsa::Error error = d_socket_sp->setOption(option);
+    return error;
+}
+
+ntsa::Error StreamSocket::startTimestampOutgoingData()
+{
+    ntsa::Error error;
+
+    if (d_timestampOutgoingData) {
+        return error;
+    }
+
+    ntcs::ObserverRef<ntci::Reactor> reactorRef(&d_reactor);
+    if (!reactorRef->supportsNotifications()) {
+        error = ntsa::Error::e_NOT_IMPLEMENTED;
+        return error;
+    }
+
+    d_totalBytesSent        = 0;
```

Comment:
Well, technically we can work with the only d_totalBytesSent without any reset of the variable because we do not support dynamic turning on of the transmit timestamping. But if in the future we decide to target more advanced kernel then it will be possible to implement and a separate variable will be required.

So I agree, I'll implement separate counters.

Reply:
Done